tests/test_horizons.cpp

```diff
@@ -0,0 +1,993 @@
+    auto result = answer.find("result");
+    CHECK(result != answer.end());
+
+    HorizonsFile horizonsFile(filePath, *result);
+    HorizonsResultCode resultCode = isValidHorizonsFile(horizonsFile.file());
+    CHECK(resultCode == expectedResultCode);
+
+    CHECK(std::filesystem::is_regular_file(filePath));
+    if (shouldDeleteFile) {
+        std::filesystem::remove(filePath);
+        CHECK(!std::filesystem::is_regular_file(filePath));
+    }
+}
+
+void testReadingHorizons(HorizonsType type, std::filesystem::path filePath,
+    double t0, double x0, double y0, double z0,
+    double t1, double x1, double y1, double z1,
+    double t2, double x2, double y2, double z2)
+{
+    // Get files and make sure they exist
+    std::filesystem::path kernel = absPath("${BASE}/tests/horizonsTest/naif0012.tls");
+    CHECK(std::filesystem::is_regular_file(kernel));
+    CHECK(std::filesystem::is_regular_file(filePath));
+
+    // Initialize SpiceManager and load leap second kernel
+    SpiceManager::initialize();
+    openspace::SpiceManager::ref().loadKernel(kernel.string());
+
+    // Read the file
+    HorizonsResult result = readHorizonsFile(filePath);
+
+    // Check the result
+    CHECK(result.type == type);
+    CHECK(result.errorCode == HorizonsResultCode::Valid);
+
+    std::vector<HorizonsKeyframe> data = result.data;
+    CHECK(data.size() == 3);
+
+    CHECK(data[0].time == Approx(t0));
+    CHECK(data[0].position.x == Approx(x0));
+    CHECK(data[0].position.y == Approx(y0));
+    CHECK(data[0].position.z == Approx(z0));
+
+
+    CHECK(data[1].time == Approx(t1));
+    CHECK(data[1].position.x == Approx(x1));
+    CHECK(data[1].position.y == Approx(y1));
+    CHECK(data[1].position.z == Approx(z1));
+
+
+    CHECK(data[2].time == Approx(t2));
+    CHECK(data[2].position.x == Approx(x2));
+    CHECK(data[2].position.y == Approx(y2));
+    CHECK(data[2].position.z == Approx(z2));
+
+    // Clean up
+    openspace::SpiceManager::deinitialize();
+}
+#endif // OPENSPACE_MODULE_SPACE_ENABLED
+
+// Test if the space module is enable or not
+TEST_CASE("HorizonsFile: Space module", "[horizonsfile]") {
+#ifdef OPENSPACE_MODULE_SPACE_ENABLED
+    CHECK(true);
+    LINFOC("test_horizons", "Downloading Horizons test data ...");
+#else
+    LERRORC("test_horizons", "Theese tests requires the Space module to be enabled");
+    CHECK(false);
+#endif // OPENSPACE_MODULE_SPACE_ENABLED
+}
+
+// Test Horizons requests and error handling
+// Test the errors that are captured by the error field in the json result
+TEST_CASE("HorizonsFile: File size too large", "[horizonsfile]") {
+#ifdef OPENSPACE_MODULE_SPACE_ENABLED
+    std::string observer = "@ssb";
+    std::string target = "-74"; // MRO
+    std::string start = "2005-08-13 00:00:00";
+    std::string stop = "2022-07-01 00:00:00";
+    std::string step = "1";
+    std::string unit = "m";
+    std::filesystem::path filePath = absPath("${BASE}/tests/horizonsTest/horizonstest_1.hrz");
```

Comment:
Another idea similar to what Alex is suggesting is to create structs here, like combining time variables into a single struct: `{start, stop, step, unit}`.